src/vsphere_cpi/lib/cloud/vsphere/vm_config.rb

```diff
@@ -122,7 +135,9 @@ def config_spec_params
       params = {}
       params[:num_cpus] = vm_type.cpu
       params[:memory_mb] = vm_type.ram
+      params[:cpu_allocation] = VimSdk::Vim::ResourceAllocationInfo.new(reservation: @maxMhz * vm_type.cpu) if vm_type.cpu_reserve_full_mhz
       params[:nested_hv_enabled] = true if vm_type.nested_hardware_virtualization
+      params[:memory_reservation_locked_to_max] = true if vm_type.memory_reservation_locked_to_max     
```

Comment:
Does this have any bearing on resource utilization calculation for picking a vSphere cluster within an AZ w.r.t to free memory? Hard reservations might change the way we calculate but not too sure.
 
We don't account for CPU utilization for cluster selection also. Will this amplify any imbalance and should we account for CPU in cluster selection?

Reply:
The memory calculation actually fits with this:
- it uses the memory available on the entire cluster , or looks per-host in the case of a MUST rule host-group
- if the memory is available, that means it is available to be locked.  primarily this setting prevents ballooning or swapping

With regards to CPU, the BOSH CPI is delegating to vsphere DRS to find the right host with the requirements.   When setting a CPU reservation, DRS will error the `create_VM` call if there isn't enough clock cycles available on the host.  

This one is a bit more possible to error out from DRS because we aren't calculating CPU capacity at all, but I'm not sure it matters that much (if `create_vm` fails on a given cluster, the `vm_creator` will just swallow the exception and try the next cluster, no?)